android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/CloudRepository.java

```diff
@@ -124,6 +124,20 @@ private boolean shouldUseCache(Context context, File cacheFile) {
     }
   }
 
+  public boolean hasAssociatedLexicalModel(@NonNull Context context, String languageID) {
+    boolean status = false;
+
+    if (memCachedDataset != null) {
+      for (int i=0; i < memCachedDataset.lexicalModels.getCount(); i++) {
+        LexicalModel lm = memCachedDataset.lexicalModels.getItem(i);
+        if (lm.getLanguageID().equalsIgnoreCase(languageID)) {
+          return true;
+        }
+      }
+    }
+    return status;
```

Comment:
This should just be `return false;` and get rid of the unnecessary `status` variable.